File: shared/models/service_result.py
# AI-SUMMARY: 标准响应包装（Python）：成功/错误响应格式
# 对应 INDEX.md §9 文件摘要索引

"""统一服务返回模型。"""

from __future__ import annotations

from typing import Any, Dict

from shared.time.shanghai_time import now_iso


def normalize_error(error: Any) -> str:
    """把异常对象统一转换成人可读文本。"""

    if error is None:
        return "Unknown error"
    if isinstance(error, str):
        return error
    return str(getattr(error, "message", None) or error)


def build_success(data: Any = None, **extra: Any) -> Dict[str, Any]:
    """构造统一成功返回结构。"""

    payload = {
        "success": True,
        "data": data,
        "error": None,
        "updateTime": extra.pop("updateTime", now_iso()),
    }
    payload.update(extra)
    return payload


def build_error(error: Any, data: Any = None, **extra: Any) -> Dict[str, Any]:
    """构造统一失败返回结构。"""

    payload = {
        "success": False,
        "data": data,
        "error": normalize_error(error),
        "updateTime": extra.pop("updateTime", now_iso()),
    }
    payload.update(extra)
    return payload

File: shared/models/service_result.js
// AI-SUMMARY: 标准响应包装（JS）：成功/错误响应格式
// 对应 INDEX.md §9 文件摘要索引

"use strict";

/**
 * 统一服务返回模型。
 * 目标是让 Node 侧的服务层都返回相同的 success / data / error 结构。
 */

function nowIso() {
  return new Date().toISOString();
}

function normalizeError(error) {
  if (!error) return "Unknown error";
  if (typeof error === "string") return error;
  return error instanceof Error ? error.message || "Unknown error" : String(error);
}

function buildSuccess(data = null, extra = {}) {
  return {
    success: true,
    data,
    error: null,
    updateTime: extra.updateTime || nowIso(),
    ...extra,
  };
}

function buildError(error, data = null, extra = {}) {
  return {
    success: false,
    data,
    error: normalizeError(error),
    updateTime: extra.updateTime || nowIso(),
    ...extra,
  };
}

function isServiceResult(value) {
  return Boolean(value && typeof value === "object" && Object.prototype.hasOwnProperty.call(value, "success"));
}

module.exports = {
  nowIso,
  normalizeError,
  buildSuccess,
  buildError,
  isServiceResult,
};
